src/clients/c/tb_client.zig

```diff
@@ -52,131 +53,199 @@ fn client_to_context(tb_client: tb_client_t) *ContextImplementation {
+            addresses,
+            packets_count,
+            on_completion_ctx,
+            on_completion_fn,
+        ) catch |err| return context_error_to_status(err);
+
+        out_client.* = client;
+        return .success;
+    }
+};
+
 // Only export the functions if we're compiling libtb_client.
 // If it's only being imported by another zig file, then exporting the functions
 // will force them to be evaluated/codegen/linked and trigger unexpected comptime paths.
 comptime {
     if (builtin.link_libc) {
-        @export(tb_client_init, .{ .name = "tb_client_init", .linkage = .Strong });
-        @export(tb_client_init_echo, .{ .name = "tb_client_init_echo", .linkage = .Strong });
-        @export(tb_client_acquire_packet, .{ .name = "tb_client_acquire_packet", .linkage = .Strong });
-        @export(tb_client_release_packet, .{ .name = "tb_client_release_packet", .linkage = .Strong });
-        @export(tb_client_submit, .{ .name = "tb_client_submit", .linkage = .Strong });
-        @export(tb_client_deinit, .{ .name = "tb_client_deinit", .linkage = .Strong });
+        @export(ffi.init, .{ .name = "tb_client_init", .linkage = .Strong });
+        @export(ffi.init_echo, .{ .name = "tb_client_init_echo", .linkage = .Strong });
+        @export(acquire_packet, .{ .name = "tb_client_acquire_packet", .linkage = .Strong });
+        @export(release_packet, .{ .name = "tb_client_release_packet", .linkage = .Strong });
+        @export(submit, .{ .name = "tb_client_submit", .linkage = .Strong });
+        @export(deinit, .{ .name = "tb_client_deinit", .linkage = .Strong });
     }
 }
 
-pub fn tb_client_init(
-    out_client: *tb_client_t,
+pub fn init(
+    allocator: std.mem.Allocator,
     cluster_id: u32,
-    addresses_ptr: [*:0]const u8,
-    addresses_len: u32,
+    addresses: []const u8,
     packets_count: u32,
     on_completion_ctx: usize,
     on_completion_fn: tb_completion_t,
-) callconv(.C) tb_status_t {
-    return init(
+) InitError!tb_client_t {
+    return init_with_context(
         DefaultContext,
-        out_client,
+        allocator,
         cluster_id,
-        addresses_ptr,
-        addresses_len,
+        addresses,
         packets_count,
         on_completion_ctx,
         on_completion_fn,
     );
 }
 
-pub fn tb_client_init_echo(
-    out_client: *tb_client_t,
+pub fn init_echo(
+    allocator: std.mem.Allocator,
     cluster_id: u32,
-    addresses_ptr: [*:0]const u8,
-    addresses_len: u32,
+    addresses: []const u8,
     packets_count: u32,
     on_completion_ctx: usize,
     on_completion_fn: tb_completion_t,
-) callconv(.C) tb_status_t {
-    return init(
+) InitError!tb_client_t {
+    return init_with_context(
         TestingContext,
-        out_client,
+        allocator,
         cluster_id,
-        addresses_ptr,
-        addresses_len,
+        addresses,
         packets_count,
         on_completion_ctx,
         on_completion_fn,
     );
 }
 
-fn init(
+fn init_with_context(
```

Comment:
I think it's ok to repeat the pattern `Context.init( ... ) catch |err| return context_error_to_status(err);`. We don't need an `init_with_context` function.